pkg/controllers/metrics/pod/controller.go

```diff
@@ -0,0 +1,194 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pod
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"knative.dev/pkg/logging"
+
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+	"github.com/prometheus/client_golang/prometheus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	crmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	podName             = "name"
+	podNameSpace        = "namespace"
+	ownerSelfLink       = "owner"
+	podHostName         = "node"
+	podProvisioner      = "provisioner"
+	podHostZone         = "zone"
+	podHostArchitecture = "arch"
+	podHostCapacityType = "capacitytype"
+	podHostInstanceType = "instancetype"
+	podPhase            = "phase"
+	podLabels           = "podLabels"
+)
+
+var (
+	podGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "karpenter",
+			Subsystem: "pods",
+			Name:      "podstate",
```

Comment:
pod_state. Maybe simply to just name this pod.

Reply:
Does this stutter with pods? pods_podstate? Maybe we can do karpenter_pod_state_X

Reply:
I modified it to be karpenter_pods_state.